### PR TITLE
Add COPY ON SEGMENT

### PR DESCRIPTION
**Pull request by janebeckman**
Usage for COPY ON SEGMENT functionality.

### CODE REVIEW

File: gpdb-doc/dita/ref_guide/sql_commands/COPY.xml [hunk 1]
```diff
@@ -37,19 +38,34 @@ COPY {table [(<varname>column</varname> [, ...])] | (<varname>query</varname>)}
...
+        which remain on the segment hosts. The <varname>file</varname> argument for <codeph>ON
+          SEGMENT</codeph> takes the string literal <codeph>&lt;SEGID></codeph> (required) and uses
+        either the absolute path or the <codeph>&lt;SEG_DATA_DIR></codeph> string literal. When the
+          <codeph>COPY</codeph> operation is run, the segment IDs and the paths of the segment data
+        directories are substituted for the string literal values. </p>
+      <p>The <codeph>ON SEGMENT</codeph> parameter allows you to copy table data to files on
+        segments for use in operations such as migrating data between clusters or performing a
+        backup. Segment data created by the <codeph>ON SEGMENT</codeph> parameter can be restored by
+        tools such as <codeph>gpfdist</codeph>, which is useful for high speed data loading.
+            <note><codeph>COPY FROM</codeph> is not currently supported.</note></p>
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/COPY.xml]: This note needs to be more specific. Something like:  COPY FROM does not support copying data from the segment files that 
 COPY TO generates.

File: gpdb-doc/dita/ref_guide/sql_commands/COPY.xml [hunk 2]
```diff
@@ -115,6 +131,31 @@ COPY {table [(<varname>column</varname> [, ...])] | (<varname>query</varname>)}
           <pd>Specifies that output goes to the client application. </pd>
         </plentry>
         <plentry>
+          <pt>ON SEGMENT</pt>
+          <pd>Copy table data to files on segments, then copy from those files back into the
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/COPY.xml]: ", then copy from those files back into the segment." doesn't seem correct.  Isn't that what COPY FROM would do if it were supported?

File: gpdb-doc/dita/ref_guide/sql_commands/COPY.xml [hunk 3]
```diff
@@ -115,6 +131,31 @@ COPY {table [(<varname>column</varname> [, ...])] | (<varname>query</varname>)}
...
+            segment. Currently does not support <codeph>COPY FROM</codeph> or <codeph>COPY TO
+              STDOUT</codeph>. The <codeph>&lt;SEG_DATA_DIR></codeph> and
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/COPY.xml]: Can't refer to these string literal values as parameters, as that suggests the values need to be replaced.

File: gpdb-doc/dita/ref_guide/sql_commands/COPY.xml [hunk 4]
```diff
@@ -115,6 +131,31 @@ COPY {table [(<varname>column</varname> [, ...])] | (<varname>query</varname>)}
...
+              <codeph>&lt;SEGID></codeph> parameters are used with <codeph>ON SEGMENT</codeph>.
+            Mirror segments are not copied.<parml>
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/COPY.xml]: Might want to rephrase this as "Mirror segments do not copy their data to files."

File: gpdb-doc/dita/ref_guide/sql_commands/COPY.xml [hunk 5]
```diff
@@ -115,6 +131,31 @@ COPY {table [(<varname>column</varname> [, ...])] | (<varname>query</varname>)}
...
+                  path(s) when <codeph>COPY</codeph> is run. (Optional. An absolute path can be used
+                  in place of using the <codeph>&lt;SEG_DATA_DIR></codeph> string literal.)</pd>
+              </plentry>
+            </parml><parml>
+              <plentry>
+                <pt>&lt;SEGID></pt>
+                <pd>The string literal representing the content ID number of the segment to be
+                  copied when copying <codeph>ON SEGMENT</codeph>. Brackets are part of string
+                  literal used to specify the path. <codeph>COPY</codeph> replaces the string
+                  literal with the content ID when <codeph>COPY</codeph> is run.</pd>
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/COPY.xml]: Should probably add that "The &lt;SEGID&gt; string literal is required as part of the ON SEGMENT option."

File: gpdb-doc/dita/ref_guide/sql_commands/COPY.xml [hunk 6]
```diff
@@ -502,6 +543,41 @@ COPY {table [(<varname>column</varname> [, ...])] | (<varname>query</varname>)}
         isolation mode and log errors:</p>
       <codeblock>COPY sales FROM '/home/usr1/sql/sales_data' LOG ERRORS 
    SEGMENT REJECT LIMIT 10 ROWS;</codeblock>
+      <p>To copy segment data for later use, use the <codeph>ON SEGMENT</codeph> argument. <note>The
+            <codeph>COPY FROM</codeph> operation does not currently support the <codeph>ON
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/COPY.xml]: Probably better to move this note about COPY FROM restriction to the end of this example, and include suggestion of using gpfdist for processing the files.

File: gpdb-doc/dita/ref_guide/sql_commands/COPY.xml [hunk 7]
```diff
@@ -37,19 +38,34 @@ COPY {table [(<varname>column</varname> [, ...])] | (<varname>query</varname>)}
...
-        the connection between the client and the master.</p>
+        the name must be specified from the viewpoint of the master host. </p>
+      <p>When <codeph>COPY</codeph> is used with the <codeph>ON SEGMENT</codeph> option, the
+          <codeph>COPY TO</codeph> causes segments to create individual segment-oriented files,
+        which remain on the segment hosts. The <varname>file</varname> argument for <codeph>ON
+          SEGMENT</codeph> takes the string literal <codeph>&lt;SEGID></codeph> (required) and uses
+        either the absolute path or the <codeph>&lt;SEG_DATA_DIR></codeph> string literal. When the
+          <codeph>COPY</codeph> operation is run, the segment IDs and the paths of the segment data
+        directories are substituted for the string literal values. </p>
+      <p>The <codeph>ON SEGMENT</codeph> parameter allows you to copy table data to files on
```
Comment (mkiyama) [inline gpdb-doc/dita/ref_guide/sql_commands/COPY.xml]: I think this should be "ON SEGMENT option"

File: gpdb-doc/dita/ref_guide/sql_commands/COPY.xml [hunk 8]
```diff
@@ -115,6 +131,31 @@ COPY {table [(<varname>column</varname> [, ...])] | (<varname>query</varname>)}
           <pd>Specifies that output goes to the client application. </pd>
         </plentry>
         <plentry>
+          <pt>ON SEGMENT</pt>
+          <pd>Copy table data to files on segments, then copy from those files back into the
+            segment. Currently does not support <codeph>COPY FROM</codeph> or <codeph>COPY TO
+              STDOUT</codeph>. The <codeph>&lt;SEG_DATA_DIR></codeph> and
+              <codeph>&lt;SEGID></codeph> parameters are used with <codeph>ON SEGMENT</codeph>.
+            Mirror segments are not copied.<parml>
+              <plentry>
```
Comment (mkiyama) [inline gpdb-doc/dita/ref_guide/sql_commands/COPY.xml]: Showing the syntax that is used with ON SEGMENT might help give the user more context about usage.

File: gpdb-doc/dita/ref_guide/sql_commands/COPY.xml [hunk 9]
```diff
@@ -42,8 +42,9 @@ COPY {table [(<varname>column</varname> [, ...])] | (<varname>query</varname>)}
         based on the segment ID if copying <codeph>ON SEGMENT</codeph>), while <codeph>COPY
           FROM</codeph> copies data from a file to a table (appending the data to whatever is in the
         table already). <codeph>COPY TO</codeph> can also copy the results of a
-          <codeph>SELECT</codeph> query. <note>Use of <codeph>COPY FROM</codeph> with the <codeph>ON
-            SEGMENT</codeph> option is not yet supported.</note></p>
+          <codeph>SELECT</codeph> query. <note><codeph>COPY FROM</codeph> the segment files
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/COPY.xml]: This note is a bit difficult to read.  Better to re-use the second note from line 65.
